src/consensus/ledgerenclavetypes.h

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "ds/ringbuffer_types.h"
+
+namespace consensus
```

Comment:
Thinking about it, I am not sure this should be under `consensus`. It is used to fetch entries from the ledger (e.g. `nodestate.h`) in places that are not related to the consensus. Could we move it to `node/`?

Reply:
I am not 100% convinced because they are largely used by the consensus to write/read from the ledger, even if it used by the nodestate.h, and it makes it clear that the consensus is the one that writes to the ledger